pkg/apiserver/registry/groupmember/rest.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupmember
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"antrea.io/antrea/pkg/apis/controlplane"
+	"antrea.io/antrea/pkg/apiserver/registry/networkpolicy"
+)
+
+type REST struct {
+	querier groupMembershipQuerier
+}
+
+var (
+	_ rest.Storage           = &REST{}
+	_ rest.Scoper            = &REST{}
+	_ rest.GetterWithOptions = &REST{}
+)
+
+// NewREST returns a REST object that will work against API services.
+func NewREST(querier groupMembershipQuerier) *REST {
+	return &REST{querier}
+}
+
+type groupMembershipQuerier interface {
+	GetGroupMembersByNamespace(name, namespace string) (controlplane.GroupMemberSet, []controlplane.IPBlock, error)
+}
+
+func (r *REST) New() runtime.Object {
+	return &controlplane.GroupMembers{}
+}
+
+func (r *REST) Destroy() {
+}
+
+func (r *REST) Get(ctx context.Context, name string, options runtime.Object) (runtime.Object, error) {
+	ns, ok := request.NamespaceFrom(ctx)
+	if !ok || len(ns) == 0 {
+		return nil, errors.NewBadRequest("Namespace parameter required.")
+	}
+	groupMembers, ipBlocks, err := r.querier.GetGroupMembersByNamespace(name, ns)
+	if err != nil {
+		return nil, errors.NewInternalError(err)
+	}
+	// Retrieve options used for pagination.
+	getOptions, ok := options.(*controlplane.PaginationGetOptions)
```

Comment:
If we use a shared type for Group and ClusterGroup, we can also share the Get() code with ClusterGroup?

Reply:
I think we could extract the common code even when there are two different structs. It just needs a function like `GetPaginatedMembers(querier groupMembershipQuerier, name string, options runtime.Object) (members []GroupMember, ipNets []IPNet, totalMembers, totalPages, currentPage int64)`